Adds a little program to check potential space savings in symbol trie (#60703)
sourcegraph/sourcegraph: dev/codeintel-merge-tries/README.md

A little program to check how many rows we could save by merging tries inside our
`codeintel_scip_symbol_names` table.

> After uploading an index to your local sourcegraph instance, run like so:
> `psql -d sourcegraph -c "COPY (select * from codeintel_scip_symbol_names where upload_id = (select max(upload_id) from codeintel_scip_symbol_names)) TO STDOUT DELIMITER ',' CSV HEADER;" | cargo run --release`
> 
> For example when running after uploading an index for sourcegraph/sourcegraph:

Reviewer: ```suggestion
Upload an index to your local Sourcegraph instance and run:
```

(Since the command is redundant with what is shown below.)

Author: Thanks, I split out the command itself into its own block so it's easier to copy (GitHub has this copy button for code blocks)

```

> $ psql -d sourcegraph -c "COPY (select * from codeintel_scip_symbol_names where upload_id = (select max(upload_id) from codeintel_scip_symbol_names)) TO STDOUT DELIMITER ',' CSV HEADER;" | cargo run --release

Reviewer: ```suggestion
$ psql --no-psqlrc --csv --dbname=sourcegraph --command="select * from codeintel_scip_symbol_names where upload_id = (select max(upload_id) from codeintel_scip_symbol_names)" | cargo run --release
```

Slightly cleaner with longer flags for documentation
